src/p5xr/p5vr/p5vr.js

```diff
@@ -49,46 +46,67 @@ export default class p5vr extends p5xr {
       window.setup();
       p5.instance._millisStart = window.performance.now();
     }
+    const refSpaceRequest = this.isImmersive ? 'local' : 'viewer';
+      this.xrSession.requestReferenceSpace(refSpaceRequest)
+        .then((refSpace) => {
+          this.xrRefSpace = refSpace;
+          // Inform the session that we're ready to begin drawing.
+          this.xrSession.requestAnimationFrame(this.onXRFrame.bind(this));
+          if (!this.isImmersive) {
+            this.xrSession.updateRenderState({
+              baseLayer: new XRWebGLLayer(this.xrSession, this.gl),
+              inlineVerticalFieldOfView: 90 * (Math.PI / 180),
+            });
+            this.addInlineViewListeners(this.canvas);
+          }
+        });
     this.onRequestSession();
   }
 
   /**
-   * `device.requestSession()` must be called within a user gesture event.
+   * `navigator.xr.requestSession('immersive-vr')` must be called within a user gesture event.
    * @param {XRDevice}
    */
   onXRButtonClicked() {
-    if (this.isImmersive) {
-      console.log('requesting session with mode: immersive-vr');
+    if (this.hasImmersive) {
+      console.log('Requesting session with mode: immersive-vr');
+      this.isImmersive = true; 
+      this.xrDevice = null;
+      this.xrSession = null;
+      this.xrRefSpace = null;
+      this.xrViewerSpace = null;
+      this.xrHitTestSource = null;
+      this.gl = null;
+      this.frame = null;
       navigator.xr.requestSession('immersive-vr').then(this.startSketch.bind(this));
     } else {
-      console.log('requesting session with mode: non-immersive-vr');
       this.xrButton.hide();
-      // Start up an inline session, which should always be supported on
-      // browsers that support WebXR regardless of the available hardware.
-      navigator.xr.requestSession('inline').then(this.startSketch.bind(this));
+      //TODO: Request Fullscreen
     }
   }
 
   onRequestSession() {
-    this.xrButton.setTitle(this.isVR ? 'EXIT VR' : 'EXIT AR');
+    // this.xrButton.setTitle(this.isVR ? 'EXIT VR' : 'EXIT AR');
```

Comment:
This can be removed. Better to get rid of stale code and if this functionality returns down the road we can start afresh.

Reply:
Removed with this commit: 02aba6760472cb0ea9cd759f53a446f3c691026c

Reply:
Removed 9302cae46427df9c7e6a86157b05febab3d11f4d